diff --git a/drw.c b/drw.c
index 8fd1ca4..042a434 100644
--- a/drw.c
+++ b/drw.c
@@ -287,7 +287,28 @@ drw_text(Drw *drw, int x, int y, unsigned int w, unsigned int h, unsigned int lp
 		nextfont = NULL;
 		while (*text) {
 			utf8charlen = utf8decode(text, &utf8codepoint, UTF_SIZ);
-			for (curfont = drw->fonts; curfont; curfont = curfont->next) {
+			curfont = drw->fonts;
+			switch (utf8codepoint) {
+				case 0x2000:
+				case 0x2001:
+				case 0x2002:
+				case 0x2003:
+				case 0x2004:
+				case 0x2005:
+				case 0x2006:
+				case 0x2007:
+				case 0x2008:
+				case 0x2009:
+				case 0x200A:
+				case 0x200B:
+				case 0x200C:
+				case 0x200D:
+				case 0x200E:
+				case 0x200F:
+					curfont = curfont->next;
+					break;
+			}
+			for (; curfont; curfont = curfont->next) {
 				charexists = charexists || XftCharExists(drw->dpy, curfont->xfont, utf8codepoint);
 				if (charexists) {
 					if (curfont == usedfont) {
